docs/kop.md

```diff
@@ -0,0 +1,135 @@
+---
+dockerfile: "https://hub.docker.com/r/streamnative/kop(TBD)"
+---
+
+KoP (Kafka on Pulsar) brings the native Apache Kafka protocol support to Apache Pulsar by introducing a Kafka protocol handler on Pulsar brokers. By adding the KoP protocol handler to your existing Pulsar cluster, you can migrate your existing Kafka applications and services to Pulsar without modifying the code. This enables Kafka applications to leverage Pulsar’s powerful features, such as:
+
+- Streamlined operations with enterprise-grade multi-tenancy
+- Simplified operations with a rebalance-free architecture
+- Infinite event stream retention with Apache BookKeeper and tiered storage
+- Serverless event processing with Pulsar Functions
+
+KoP, implemented as a Pulsar [protocol handler](https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/protocol/ProtocolHandler.java) plugin with the protocol name "kafka", is loaded when Pulsar broker starts. It helps reduce the barriers for people adopting Pulsar to achieve their business success by providing a native Kafka protocol support on Apache Pulsar. By integrating the two popular event streaming ecosystems, KoP unlocks new use cases. You can leverage advantages from each ecosystem and build a truly unified event streaming platform with Apache Pulsar to accelerate the development of real-time applications and services.
+
+KoP implements the Kafka wire protocol on Pulsar by leveraging the existing components (such as topic discovery, the distributed log library - ManagedLedger, cursors and so on) that Pulsar already has.
+
+The following figure illustrates how the Kafka-on-Pulsar protocol handler is implemented within Pulsar.
+
+![](docs/kop-architecture.png)
```

Comment:
the image is not shown on preview, seems that you do not commit this image

Reply:
it's already in this repo